src/components/BBCode/BBCode.tsx

```diff
@@ -1,15 +1,66 @@
-import React, { FC } from 'react';
-
-import BBCodeUntyped from '@bbob/react/es/Component';
-import reactPreset from '@bbob/preset-react/es';
-
-import cls from './BBCode.module.css';
-
-/**
- * Typed BBCode component.
- */
-export const BBCode: FC = ({ children }) => (
-  <BBCodeUntyped plugins={[reactPreset()]} componentProps={{ className: cls.bbcode }}>
-    {children}
-  </BBCodeUntyped>
-);
+import React, { FC } from 'react';
+
+import { parse, ast, ast_item, ast_to_array } from '@prekel/rescript-bbcode/src/BBCode.gen';
+import { Link } from '@chakra-ui/react';
+
+/**
+ * Transforms AST to React element array through astItemToElement function.
+ * @param a AST.
+ * @returns React element array.
+ */
+// eslint-disable-next-line @typescript-eslint/no-use-before-define
+const astToElement = (a: ast): JSX.Element[] => ast_to_array(a).map(astItemToElement);
+
+/**
+ * Transforms AST item to corresponding React element.
+ * @param astItem AST item.
+ * @returns React element.
+ */
+const astItemToElement = (astItem: ast_item): JSX.Element => {

```

Comment:
I suggest you rewrite the function like this:

```js
/**
 * Transforms AST to React element array through astItemToElement function.
 * @param a AST.
 * @returns React element array.
 */
const astToElement = (a: ast): ReactElement[] => {

  /**
   * Transforms AST item to corresponding React element.
   * @param astItem AST item.
   * @param index Item index.
   */
  const astItemToElement = (astItem: ast_item, index: number): ReactElement => {
    const itemKey = astItem.tag + index.toString();

    switch (astItem.tag) {
      case 'Text': return <span key={itemKey}>{astItem.value}</span>;
      case 'Bold': return <b key={itemKey}>{astToElement(astItem.value.children)}</b>;
      case 'Italic': return <i key={itemKey}>{astToElement(astItem.value.children)}</i>;
      case 'Link': return <Link key={itemKey} href={astItem.value.url} isExternal />;
      case 'LinkNamed':
        return <Link key={itemKey} href={astItem.value.url} isExternal>{astToElement(astItem.value.children)}</Link>;
      default:
        return <div key={itemKey}>{astItem}</div>;
    }
  };

  return ast_to_array(a).map((item, index) => astItemToElement(item, index));
};
```

This way you can avoid ignoring eslint rule and pass safer key (tag + index) rather than tag names

Reply:
Index is OK here because will always be same